### ISSUE TITLE
Raster vectorizer

### ISSUE THREAD

**Issue by msbarry** [onthegomap/planetiler#551]
There are a few applications where having a high-performance raster vectorizer will be necessary: 

- #12
- #550 
- and possibly also an alternative way to merge overlapping polygons (rasterize -> dilate -> erode -> vectorize)

**Comment by wipfli** [comment]
Great idea, would be amazing to have for stuff like https://github.com/wipfli/swisstopo-vector-hillshade...

**Comment by msbarry** [comment]
Yeah I started pulling in gdal java bindings for this kind of stuff, but I realized everything I needed from it is really just a variation of vectorizing raster datasets... so if we can get this implemented it opens the door to a lot of other applications

**Comment by dBitech** [comment]
I'm no Java expert but leaving this here for idea tracking purposes: https://github.com/geotools/geotools/blob/main/modules/unsupported/process-raster/src/main/java/org/geotools/process/raster/PolygonExtractionProcess.java
